hack/test-canary.sh

```diff
@@ -95,13 +96,34 @@ function run() {
 
 # Helper function to clean up log and json files.
 function clean() {
-    kubectl get pods --all-namespaces | grep ccm | awk '{print $1}' | xargs kubectl delete ns
-    rm "${TEST_DIR}/${TEST_PREFIX}*"
+    echo "ensuring fresh \$START."
+    unset START
+    echo "ensuring fresh ${TEST_LOG} file."
+    rm -f "${TEST_LOG}"
+    echo "ensuring fresh ${METRICS_FILE} result file."
+    rm -f  "${METRICS_FILE}" 
+    echo "ensuring all 'cm-e2e-tests' namespaces are terminated."
+    local res=$(kubectl get ns | grep 'cm-e2e-tests-' | awk '{print $1}')
+    if [ ! -z "${res}" ]; then
+        echo ${res} | xargs kubectl delete ns 2> /dev/null
+    fi
+}
+
+# Run the tests in loop with the specified wait period.
+function monitor() {
```

Comment:
The new 'monitor' method just sits in an infinite loop and calls the original 'run-once' method on each iteration. It also sleeps for the specified period. This will give a window of opportunity for any sidecars to grab the results from filesystem. They will be deleted on the next iteration of the tests and the current results lost.